modules/renter/siafile/siafile.go

```diff
@@ -0,0 +1,181 @@
+package siafile
+
+import (
+	"bytes"
+	"encoding/base32"
+	"encoding/binary"
+	"fmt"
+	"sync"
+
+	"github.com/NebulousLabs/Sia/modules"
+	"github.com/NebulousLabs/Sia/types"
+	"github.com/NebulousLabs/fastrand"
+
+	"github.com/NebulousLabs/Sia/crypto"
+)
+
+type (
+	// SiaFile is the disk format for files uploaded to the Sia network.  It
+	// contains all the necessary information to recover a file from its hosts and
+	// allows for easy constant-time updates of the file without having to read or
+	// write the whole file.
+	SiaFile struct {
```

Comment:
we should be able to make a bunch of the fields static for both the SiaFile and the Metadata, right? then we can minimize lock contention when working with the files, something that for some situations may end up being fairly serious.

Reply:
Yes, I wanted to do that once everything else in this PR looks good to you and won't be changed anymore.